### PR TITLE
GraphQL: complete 'temporary' "File2" -> "File" rename (5 months old)

### PR DESCRIPTION
**Pull request by slimsag**
We've had this comment around for 5+ months:

```
INTERNAL: This is temporarily named File2 during a migration. Do not refer to the name File2 in
any API clients as the name will change soon.
```

Let's finish this renaming now :)

Signed-off-by: Stephen Gutekanst <stephen@sourcegraph.com>

### CODE REVIEW

**Comment by eseliger** [comment]
Correction: This comment was there at least since I joined, so almost two years 🤣

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -4017,10 +4017,8 @@ In a future version of Sourcegraph, a repository's files may be distinct from a
 (for example, to support searching/browsing generated files that aren't committed and don't exist
 as Git blobs). Clients should generally use the GitBlob concrete type and GitCommit.blobs (not
 GitCommit.files), unless they explicitly want to opt-in to different behavior in the future.
-INTERNAL: This is temporarily named File2 during a migration. Do not refer to the name File2 in
-any API clients as the name will change soon.
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/schema.graphql:4021]: This is a breaking change to the API. I think one that is worthwhile, but we should at least add a small note in the changelog.

**Review by eseliger — COMMENTED** [review]
Needs fixes in the client code too

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 1 above)
Comment (efritz) [reply cmd/frontend/graphqlbackend/schema.graphql:4021]: +1

**Review by efritz — APPROVED** [review]
Approved, pending changelog note.

**Comment by slimsag** [comment]
Yikes.

I'll add a changelog entry.

**Comment by slimsag** [comment]
Gosh, our API is so complex: There's already another `type File {`... that one is ALSO deprecated 🤦

**Comment by slimsag** [comment]
!abandon - this wasn't the giant tangle of worms I was trying to get myself into today. I was just trying to fix a different issue and thought this would be a decent issue to fix as I was in the area.. but...

---

To rename `File2` -> `File`, I first need to deal with this type:

```graphql
"""
File is temporarily preserved for backcompat with browser extension search API client code.
"""
type File {
...
```

But it's still in use for search suggestions:

```
"""
A search suggestion.
"""
union SearchSuggestion = Repository | File | Symbol | Language | SearchContext
````

But `FileDiffResolver` ALSO depends on it in the graphqlbackend layer - despite claiming to use `File2` in the `schema.graphql`.

That's ignoring the fact that I would be renaming `File2` -> `File` which would conflict with the "old deprecated type File"....

**Comment by slimsag** [comment]
also - lol - `VirtualFile` is a thing and is not used apparently.

**Comment by eseliger** [comment]
I think it's used somewhere in the RepoPreviewDiff stuff, maybe it's not mentioned explicitly as a type, though

**Comment by slimsag** [comment]
😅

**Pull request closed without merging**